esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -2853,66 +2869,33 @@ void CFileSprayEx::addDropZoneFile(IEspContext& context, IDirectoryIterator* di,
     timestr.appendf("%04d-%02d-%02d %02d:%02d:%02d", y,m,d,h,min,sec);
     aFile->setModifiedtime(timestr.str());
     aFile->setFilesize(di->getFileSize());
-    if (di->isDir() && filesInFolder.ordinality())
-        aFile->setFiles(filesInFolder);
     files.append(*aFile.getLink());
 }
 
-bool CFileSprayEx::searchDropZoneFileInFolder(IEspContext& context, IFile* f, const char* nameFilter,
-    bool returnAll, const char* dropZonePath, StringBuffer& relativePath, const char pathSep, IArrayOf<IEspPhysicalFileStruct>& files)
+void CFileSprayEx::searchDropZoneFiles(IEspContext& context, IpAddress& ip, const char* dir, const char* nameFilter, IArrayOf<IEspPhysicalFileStruct>& files, unsigned& filesFound)
 {
-    bool foundMatch = false;
-    Owned<IDirectoryIterator> di = f->directoryFiles(NULL, false, true);
+    RemoteFilename rfn;
+    SocketEndpoint ep;
+    ep.ipset(ip);
+    rfn.setPath(ep, dir);
+    Owned<IFile> f = createIFile(rfn);
+    if(!f->isDirectory())
```

Comment:
Does it matter it it is a single filename?

@jakesmith I assume this will cause an extra remote call.  Is it better to call directoryFiles and catch any exception?

Reply:
It does involve an extra remote call, not sure that is overly significant though (?)

It looks like directoryFiles wouldn't throw an exception, but simply return 0 results.
And that's true in the non-remote version too..

So if must given an exception if not a directory, prob. need to keep the separate call.

Reply:
The dir should not be a single filename. If it is, there is something wrong in environment setting.